api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -43,8 +55,89 @@ data Logger = Logger
 
 {- | This is a data type that can represent logs from various sources: application log files, machine generated events, system logs, etc. [Specification outlined here.](https://opentelemetry.io/docs/specs/otel/logs/data-model/)
 Existing log formats can be unambiguously mapped to this data type. Reverse mapping from this data type is also possible to the extent that the target log format has equivalent capabilities.
+Uses an IORef under the hood to allow mutability.
 -}
-data LogRecord body = LogRecord
+data ReadWriteLogRecord = ReadWriteLogRecord Logger (IORef ImmutableLogRecord)
+
+
+mkReadWriteLogRecord :: Logger -> ImmutableLogRecord -> IO ReadWriteLogRecord
+mkReadWriteLogRecord l = fmap (ReadWriteLogRecord l) . newIORef
+
+
+newtype ReadableLogRecord = ReadableLogRecord {readableLogRecord :: ReadWriteLogRecord}
```

Comment:
Changed to be a concrete type so that it can be used in `data LogRecordExporter`.